pkg/arvo/sys/vane/ames.hoon

```diff
@@ -691,35 +691,65 @@
   --
 ::  +scry: dereference namespace
 ::
-::    The ones producing vases are expected to be used like this:
-::
-::    &tang [(sell .^(vase %a /=peer=/~zod)) ~]
-::
 ++  scry
   |=  [fur=(unit (set monk)) ren=@tas why=shop syd=desk lot=coin tyl=path]
   ^-  (unit (unit cage))
-  ?.  =(lot [%$ %da now])  ~
-  ?.  =(%$ ren)  [~ ~]
-  ?.  =([%& our] why)
-    [~ ~]
-  ?+    syd  ~
-      %peers
-    ?^  tyl  [~ ~]
+  ::  only respond for the local identity, %$ desk, current timestamp
+  ::
+  ?.  ?&  =(&+our why)
+          =([%$ %da now] lot)
+          =(%$ syd)
+      ==
+    ~
+  ::  /ax/protocol/version           @
+  ::  /ax/peers                      (map ship ?(%alien known))
+  ::  /ax/peers/[ship]               ship-state
+  ::  /ax/peers/[ship]/forward-lane  (unit lane)
+  ::  /ax/bones/[ship]               [snd=(set bone) rcv=(set bone)]
+  ::  /ax/snd-bones/[ship]/[bone]    vase
+  ::
+  ?.  ?=(%x ren)  ~
+  ?+    tyl  ~
+      [%protocol %version ~]
+    ``noun+!>(protocol-version)
+  ::
+      [%peers ~]
     :^  ~  ~  %noun
     !>  ^-  (map ship ?(%alien %known))
     (~(run by peers.ames-state) head)
   ::
-      %peer
-    ?.  ?=([@ ~] tyl)  [~ ~]
-    =/  who  (slaw %p i.tyl)
+      [%peers @ *]
+    =/  who  (slaw %p i.t.tyl)
     ?~  who  [~ ~]
     ?~  peer=(~(get by peers.ames-state) u.who)
       [~ ~]
-    ``noun+!>(u.peer)
+    ?+  t.t.tyl  [~ ~]
+      ~  ``noun+!>(u.peer)
+    ::
+        [%forward-lane ~]
+      ::TODO  (list lane) instead?
+      ::TODO  refactor wrt +send-blob to share routing logic
+      ::  find lane for u.who, or their galaxy
+      ::
+      :^  ~  ~  %noun
+      !>  ^-  (unit lane)
+      =/  ship-state  (~(get by peers.ames-state) u.who)
+      ?.  ?=([~ %known *] ship-state)
+        ~
+      =/  peer-state  +.u.ship-state
+      ?.  =(~ route.peer-state)  ::NOTE  avoid tmi
```

Comment:
Why do we need to avoid TMI here?  I don't see `route.peer-state` used anywhere else.

Reply:
It's used implicitly by the `$(peer-state +.u.next)` a couple lines down.